src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -71,19 +73,19 @@ public JobService(IamService samService,
                       MigrateConfiguration migrateConfiguration,
                       GoogleResourceConfiguration googleResourceConfiguration,
                       ApplicationContext applicationContext,
-                      KubeService kubeService,
-                      JobShutdownState jobShutdownState,
                       Migrate migrate,
                       ObjectMapper objectMapper,
                       PerformanceLogger performanceLogger) throws StairwayExecutionException {
         this.samService = samService;
         this.appConfig = appConfig;
-        this.kubeService = kubeService;
+
         this.stairwayJdbcConfiguration = stairwayJdbcConfiguration;
         this.migrateConfiguration = migrateConfiguration;
-        this.jobShutdownState = jobShutdownState;
+        this.isShutdown = new AtomicBoolean(false);
         this.migrate = migrate;
 
+        this.kubeService = new KubeService(appConfig.getPodName(), appConfig.isInKubernetes(), API_POD_FILTER);
```

Comment:
Not sure it makes sense, but would it be possible to make this constructor an injectable service, such as by creatinga a wrapper service object? That would be helpful if there's more than one place where a `KubeService` is created.

Reply:
Yes, you could. I did feel a tiny bit bad changing the two other KubeService autowires to JobService.
But only a tiny bit :)